### ISSUE TITLE
Support for mu4e 1.8.x?

### ISSUE THREAD

**Issue by ifinkelstein** [jeremy-compostella/org-msg#157]
First, thank you for a phenomenal package! But it appears that `mu4e 1.8.x` broke `org-msg`. Are there any plans to improve compatibility with newer `mu4e` releases?

**Comment by danielfleischer** [comment]
Related to that, see [discussion](https://groups.google.com/g/mu-discuss/c/V1WxQzglkZA/m/Av9vJ_BkAgAJ) in `mu` mailing list.

**Comment by danielfleischer** [comment]
As a temporary measure, I've set `(reply-to-text . (text html))` which does the following: `mu` does not exposes the HTML anymore so every email is considered text. Gnus/shr is converting HTML to text which is put in quote block which finally is rendered nicely to HTML by org-msg. In most cases, html content is recovered completely and you can either top-post or inline-post because the quoted text is editable, up to you.

**Comment by ifinkelstein** [comment]
Thanks for the tip, seems to work as a stop-gap solution with ~mu4e 1.8.9~.

**Comment by legalnonsense** [comment]
There are only a couple functions whose names need to be changed. Here is what I changed and everything is working again. I'd create a pull request but I do not really know how, and it should also check the mu version since not everyone will have updated. 

```
@@ -1239,7 +1239,7 @@ HTML emails."
 (defun org-msg-edit-mode-mu4e ()
   "Setup mu4e faces, addresses completion and run mu4e."
   (mu4e~compose-remap-faces)
-  (mu4e~start)
+  (mu4e--start)
   (when mu4e-compose-complete-addresses
     (mu4e~compose-setup-completion))
   ;; the following code is verbatim from mu4e-compse.el, mu4e-compose-mode
@@ -1261,7 +1261,8 @@ HTML emails."
   (add-hook 'message-sent-hook
             (lambda () ;;  mu4e~compose-mark-after-sending
               (setq mu4e-sent-func 'mu4e-sent-handler)
-              (mu4e~proc-sent (buffer-file-name))) nil t))
+	      (mu4e--server-sent (buffer-file-name)))))
+              ;; (mu4e~proc-sent (buffer-file-name))) nil t))
```

**Comment by benthamite** [comment]
Hi, after reinstalling Emacs and mu, I'm getting an error whenever I try to reply to a message (but not when I compose a new message), and I'm wondering if this is due to the incompatibility of `org-msg` and mu4e 1.8.x.

mu 1.8.14

GNU Emacs 30.0.50 (build 1, aarch64-apple-darwin22.1.0, NS appkit-2299.00 Version 13.0.1 (Build 22A400)) of 2023-01-30

<details><summary>Backtrace</summary>
<p>

``` emacs-lisp
Debugger entered--Lisp error: (wrong-type-argument stringp ("John Doe" :email "john.doe@gmail.com"))
  split-string(("John Doe" :email "john.doe@gmail.com") ",+ *")
  org-html-format-spec((:export-options nil :back-end #s(org-export-backend :name org-msg-html :parent html :transcoders ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block)) :options nil :filters nil :blocks nil :menu nil) :translate-alist ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block) ...) :exported-data #<hash-table eq 13/4001 0x4c3e19af> :input-buffer " *temp*-860616" :input-file nil :html-doctype "xhtml-strict" :html-container "div" :html-content-class "content" :description nil :keywords nil :html-html5-fancy nil :html-link-use-abs-url nil :html-link-home "" :html-link-up "" :html-mathjax "" :html-equation-reference-format "\\eqref{%s}" :html-postamble nil :html-preamble t :html-head "" :html-head-extra "" :subtitle nil :html-head-include-default-style nil :html-head-include-scripts nil :html-allow-name-attribute-in-anchors nil ...))
  org-html--build-pre/postamble(preamble (:export-options nil :back-end #s(org-export-backend :name org-msg-html :parent html :transcoders ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block)) :options nil :filters nil :blocks nil :menu nil) :translate-alist ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block) ...) :exported-data #<hash-table eq 13/4001 0x4c3e19af> :input-buffer " *temp*-860616" :input-file nil :html-doctype "xhtml-strict" :html-container "div" :html-content-class "content" :description nil :keywords nil :html-html5-fancy nil :html-link-use-abs-url nil :html-link-home "" :html-link-up "" :html-mathjax "" :html-equation-reference-format "\\eqref{%s}" :html-postamble nil :html-preamble t :html-head "" :html-head-extra "" :subtitle nil :html-head-include-default-style nil :html-head-include-scripts nil :html-allow-name-attribute-in-anchors nil ...))
  org-html-template(#("<div class=\"signature\" id=\"org9800559\">\n<p>\n&#x201..." 62 67 (:parent (bold (:begin 294 :end 301 :post-blank 0 :contents-begin 295 :contents-end 300 :parent (paragraph (:begin 291 :end 302 :contents-begin 291 :contents-end 302 :post-blank 0 :post-affiliated 291 :mode nil :granularity nil :parent (special-block ... #18)) #("--\n" 0 3 (:parent #18)) #4 #("\n" 0 1 (:parent #18)))) #("Pablo" 0 5 (:parent #4))))) (:export-options nil :back-end #s(org-export-backend :name org-msg-html :parent html :transcoders ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block)) :options nil :filters nil :blocks nil :menu nil) :translate-alist ((special-block . org-msg--html-special-block) (quote-block . org-msg--html-quote-block) (bold . org-html-bold) (center-block . org-html-center-block) (clock . org-html-clock) (code . org-html-code) (drawer . org-html-drawer) (dynamic-block . org-html-dynamic-block) (entity . org-html-entity) (example-block . org-html-example-block) (export-block . org-html-export-block) (export-snippet . org-html-export-snippet) (fixed-width . org-html-fixed-width) (footnote-reference . org-html-footnote-reference) (headline . org-html-headline) (horizontal-rule . org-html-horizontal-rule) (inline-src-block . org-html-inline-src-block) (inlinetask . org-html-inlinetask) (inner-template . org-html-inner-template) (italic . org-html-italic) (item . org-html-item) (keyword . org-html-keyword) (latex-environment . org-html-latex-environment) (latex-fragment . org-html-latex-fragment) (line-break . org-html-line-break) (link . org-html-link) (node-property . org-html-node-property) (paragraph . org-html-paragraph) (plain-list . org-html-plain-list) (plain-text . org-html-plain-text) (planning . org-html-planning) (property-drawer . org-html-property-drawer) (quote-block . org-html-quote-block) (radio-target . org-html-radio-target) (section . org-html-section) (special-block . org-html-special-block) (src-block . org-html-src-block) (statistics-cookie . org-html-statistics-cookie) (strike-through . org-html-strike-through) (subscript . org-html-subscript) (superscript . org-html-superscript) (table . org-html-table) (table-cell . org-html-table-cell) (table-row . org-html-table-row) (target . org-html-target) (template . org-html-template) (timestamp . org-html-timestamp) (underline . org-html-underline) (verbatim . org-html-verbatim) (verse-block . org-html-verse-block) ...) :exported-data #<hash-table eq 13/4001 0x4c3e19af> :input-buffer " *temp*-860616" :input-file nil :html-doctype "xhtml-strict" :html-container "div" :html-content-class "content" :description nil :keywords nil :html-html5-fancy nil :html-link-use-abs-url nil :html-link-home "" :html-link-up "" :html-mathjax "" :html-equation-reference-format "\\eqref{%s}" :html-postamble nil :html-preamble t :html-head "" :html-head-extra "" :subtitle nil :html-head-include-default-style nil :html-head-include-scripts nil :html-allow-name-attribute-in-anchors nil ...))
  org-export-as(org-msg-html nil nil nil nil)
  org-export-to-buffer(org-msg-html " *OrgMsg HTML Export*")
  org-msg-org-to-xml(#("#+OPTIONS: html-postamble:nil H:5 num:nil ^:{} toc..." 0 2 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 2 86 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 86 87 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 87 89 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 89 127 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 127 128 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 128 140 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 140 141 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 141 219 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 219 220 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 220 236 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 236 237 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 237 263 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 263 264 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 264 269 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 269 272 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) ...) "/Users/pablostafforini/Downloads/")
  org-msg-build(#("#+OPTIONS: html-postamble:nil H:5 num:nil ^:{} toc..." 0 2 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 2 86 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 86 87 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 87 89 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 89 127 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 127 128 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 128 140 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 140 141 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 141 219 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 219 220 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 220 236 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 236 237 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 237 263 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 263 264 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 264 269 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 269 272 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) ...))
  org-msg-export-as-html(#("#+OPTIONS: html-postamble:nil H:5 num:nil ^:{} toc..." 0 2 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 2 86 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 86 87 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 87 89 (invisible org-modern face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 89 127 (face org-meta-line read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 127 128 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) 128 140 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil) 140 141 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 141 219 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 219 220 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 220 236 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 236 237 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 237 263 (read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 263 264 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 264 269 (face org-drawer read-only nil wrap-prefix "" line-prefix "" font-lock-fontified t fontified nil isearch-open-invisible-temporary org-fold-core--isearch-show-temporary isearch-open-invisible org-fold-core--isearch-show org-fold--spec-org-fold-drawer-43688 org-fold-drawer) 269 272 (hard t read-only nil wrap-prefix "" line-prefix "" fontified nil) ...))
  #f(compiled-function (alt) #<bytecode -0x526f85b59f64a30>)(html)
  org-msg-build-alternatives((text html))
  org-msg-prepare-to-send()
  message-send(nil)
  message-send-and-exit(nil)
  funcall-interactively(message-send-and-exit nil)
  command-execute(message-send-and-exit)

```

</p>
</details>

**Comment by gajama** [comment]
Does it work if you use Emacs 29, the upcoming release, rather than bleeding edge? 

Is your org-msg up-to-date?

This commit https://github.com/jeremy-compostella/org-msg/commit/fc75f6b0e735eb58b362e677ab95053fe22115f4 of a PR I wrote is intended to fix (or, more accurately maybe, avoid) the issues org-msg had with mu4e 1.8.x and it is certainly working fine for me on Emacs 29.

**Comment by benthamite** [comment]
Thanks for your reply. I downgraded to Emacs 28 because I had [issues](https://github.com/d12frosted/homebrew-emacs-plus/issues/544) installing Emacs 29. Unfortunately, the problem persists, although the error is now slightly different: `split-string: Wrong type argument: stringp, ("pablo.stafforini@gmail.com")`. Enabling `debug-on-error` doesn't generate a backtrace. However, the relevant function appears to be `org-html-format-spec`. Instrumenting the function for Edebug shows that the error is triggered by the line `(split-string (plist-get info :email)  ",+ *")`.

**Comment by danielfleischer** [comment]
Did you upgrade mu from a previous version? did you recreate the index by `mu init` and then `mu index`?

**Comment by benthamite** [comment]
Yes, I upgraded from mu 1.6.11 to mu 1.8.14 (`brew upgrade mu`). Then ran ` mu init --maildir=~/Mail --my-address=myaddress1@gmail.com --my-address=myaddress2@gmail.com` and `mu index`. I just reindexed the database and the problem persists.

The error is always `split-string: Wrong type argument: stringp, ("from@email.com")`, with `from@email.com` replacing the email in the `from` field.

**Comment by gajama** [comment]
What version of Org mode?

**Comment by benthamite** [comment]
Org mode version 9.6.1 (release_9.6.1-201-ga6c882 @ /Users/pablostafforini/.emacs.d/straight/build/org/)

**Comment by gajama** [comment]
Ok, I updated to that version of Org, and I'm getting a bug that *might* be related.

When I reply to a message from myself, which has worked fine previously, the "To:" field is empty. This doesn't cause a crash, obviously, but does look like it could have a similar cause. That would suggest some change between Org 9.6 and 9.6.1 could be the culprit.

I can't trigger any other bug on my Linux machine though. I'll try on my MacBook later. That's running Emacs 29 as well, which I installed yesterday without issues, rather than 28, but seeing as you've had more-or-less the same problem in 28 and 30, I'd imagine you would get the same error with 29.

**Comment by danielfleischer** [comment]
Hi, I don't know if it's related but the installed `gmime` has at least one bug related to encoding contacts. Try to install an updated version via `brew install --head gmime`. Of course, you'll need to rebuild mu to use the updated gmime. Build, not install.

See discussion https://github.com/djcb/mu/issues/2461.

**Comment by benthamite** [comment]
Thank you for the suggestion! Unfortunately, this didn't work.

For reference, here's what I did:

1. Close Emacs
1. `brew install --head gmime`
2. `brew uninstall mu`
3. `git-clone https://github.com/djcb/mu`
4. `cd mu`
5. `./autogen.sh && make`
6. `sudo make install`
7. Initialize and reindex db
8. Delete `elpaca/builds/mu4e`
9. Start Emacs

**Comment by benthamite** [comment]
Okay, I fixed it. The problam was entirely unrelated to `gmime` or the latest version of `org-msg`. It turns out that, buried in my sprawling `mu4e` config, there was a short custom function added to `mu4e-compose-pre-hook` that was triggering the error. I believe I added this function back when `org-msg` stopped working (but before @gajama's commit), based in @danielfleischer [suggestion above](https://github.com/jeremy-compostella/org-msg/issues/157#issuecomment-1233791513). Frustratingly, it did not show up in the backtrace, so I overlooked it. Anyway, thanks everyone for the help.

(I didn't open the issue myself so I can't close it, but I don't think there are any other pending discussions.)
